src/eip7594/eip7594.c

```diff
@@ -615,95 +722,17 @@ C_KZG_RET verify_cell_kzg_proof_batch(
-            if (ret != C_KZG_OK) goto out;
-            blst_fr_mul(&scaled_fr, &cell_fr, &r_powers[i]);
-            size_t index = cell_indices[i] * FIELD_ELEMENTS_PER_CELL + j;
-            blst_fr_add(
-                &aggregated_column_cells[index], &aggregated_column_cells[index], &scaled_fr
-            );
-
-            /* Mark the cell as being used */
-            is_cell_used[index] = true;
-        }
-    }
-
-    ////////////////////////////////////////////////////////////////////////////////////////////////
-    // Compute sum of the interpolation polynomials
+    // Commmit to aggregated interpolation polynomial: [sum_k r^k interpolation_poly_k(s)]
     ////////////////////////////////////////////////////////////////////////////////////////////////
 
-    /* Start with a zeroed out poly */
-    for (size_t i = 0; i < FIELD_ELEMENTS_PER_CELL; i++) {
-        aggregated_interpolation_poly[i] = FR_ZERO;
-    }
-
-    /* Interpolate each column */
-    for (size_t i = 0; i < CELLS_PER_EXT_BLOB; i++) {
-        /* Offset to the first cell for this column */
-        size_t index = i * FIELD_ELEMENTS_PER_CELL;
-
-        /* We only care about initialized cells */
-        if (!is_cell_used[index]) continue;
-
-        /* We don't need to copy this because it's not used again */
-        ret = bit_reversal_permutation(
-            &aggregated_column_cells[index], sizeof(fr_t), FIELD_ELEMENTS_PER_CELL
-        );
-        if (ret != C_KZG_OK) goto out;
-
-        /*
-         * Get interpolation polynomial for this column. To do so we first do an IDFT over the roots
-         * of unity and then we scale by the coset factor.  We can't do an IDFT directly over the
-         * coset because it's not a subgroup.
-         */
-        ret = fr_ifft(
-            column_interpolation_poly, &aggregated_column_cells[index], FIELD_ELEMENTS_PER_CELL, s
-        );
-        if (ret != C_KZG_OK) goto out;
-
-        /*
-         * To unscale, divide by the coset. It's faster to multiply with the inverse. We can skip
-         * the first iteration because its dividing by one.
-         */
-        uint64_t pos = reverse_bits_limited(CELLS_PER_EXT_BLOB, i);
-        fr_t inv_coset_factor;
-        blst_fr_eucl_inverse(&inv_coset_factor, &s->roots_of_unity[pos]);
-        shift_poly(column_interpolation_poly, FIELD_ELEMENTS_PER_CELL, &inv_coset_factor);
-
-        /* Update the aggregated poly */
-        for (size_t k = 0; k < FIELD_ELEMENTS_PER_CELL; k++) {
-            blst_fr_add(
-                &aggregated_interpolation_poly[k],
-                &aggregated_interpolation_poly[k],
-                &column_interpolation_poly[k]
-            );
-        }
-    }
-
-    /* Commit to the final aggregated interpolation polynomial */
-    ret = g1_lincomb_fast(
-        &evaluation, s->g1_values_monomial, aggregated_interpolation_poly, FIELD_ELEMENTS_PER_CELL
+    /* Aggregated cells from same columns, sum interpolation polynomials, and commit */
```

Comment:
Oh one more thing. This should be:
```suggestion
    /* Aggregate cells from same columns, sum interpolation polynomials, and commit */
```